src/waveform/widgets/glwaveformwidgetabstract.h

```diff
@@ -1,30 +1,33 @@
 #pragma once
 
-#include <QGLWidget>
 #include <QString>
 
 #include "waveform/renderers/glwaveformrenderer.h"
-#include "waveform/sharedglcontext.h"
 #include "waveform/widgets/waveformwidgetabstract.h"
+#include "widget/wglwidget.h"
 
 QT_FORWARD_DECLARE_CLASS(QString)
 
-/// GLWaveformWidgetAbstract is a WaveformWidgetAbstract & QGLWidget that has
-/// a GLWaveformRenderer member which requires initialization that must be
-/// deferred until Qt calls QGLWidget::initializeGL and cannot be done in the
+/// GLWaveformWidgetAbstract is a WaveformWidgetAbstract & WGLWidget. It may
+/// have a GLWaveformRenderer member which requires initialization that must be
+/// deferred until Qt calls WGLWidget::initializeGL and cannot be done in the
```

Comment:
I will need to investigate this. But there were several widgets that derived from QGLWidget + WaveformWidgetAbstract, and other widgets that derived from GLWaveformWidgetAbstract, which derived from QGLWidget + WaveformWidgetAbstract, which made no sense to me. The only difference between the two approaches was this additional  GLWaveformRenderer member in GLWaveformWidgetAbstract and how it is called from initializeGL.

Reply:
And about the Qt 6 guard for QGLWidget: Correct me if I'm wrong, but main doesn't compile against Qt 6 either, does it? It will be easier to deal with in my branch though, as this include and the use of QGLWidget is now limited to a single location. I guess I could include and inherit from QWidget instead, so the code at least compiles (as long as all calls to QGLWidget methods are Qt 6 guarded too, which I think they are).